src/components/GroupForm.js

```diff
@@ -29,31 +31,43 @@ export default connect(mapStateToProps)(function GroupForm ({users, userId, memb
           </Item>
         </Form>
 
-        <Form>
-            <Item>
-              <Input placeholder='Search Users'/>
-            </Item>
-        </Form>
+        <Item>
+          <List dataArray={tempList}
+            renderRow={user =>
+              <ListItem>
+                <Text>{user}</Text>
+              </ListItem>
+            }>
+          </List>
+        </Item>
 
-        <List dataArray={users}
-          renderRow={user =>
-            <ListItem key={user.id} 
-            onPress={() => dispatch({type: 'ADD_MEMBER', id: user.id}) }>
-              <Text>{`${user.first_name} ${user.last_name}`}</Text>
-            </ListItem>
-          }>
-        </List>
+        <Item>
+          <Input onChangeText={text => dispatch({type: 'SEARCH_NAME', text: text})} placeholder='Add a Member'/>
+          <List dataArray={searchResults}
+            renderRow={user =>
+              <ListItem onPress={() => {
+                dispatch({type: 'ADD_MEMBER', id: user.id});
+              }}>
+                <Text>{user.username}</Text>
+              </ListItem>
+            }>
+          </List>
+        </Item>
 
       </Content>
 
       <Button
         style={{ alignSelf: 'center', marginTop: 20, marginBottom: 20 }}
-        onPress={()=> {
+        onPress={() => {
           let data = {
             name: groupName,
             members: [...members, userId ]
           };
-          postGroup(data).then(data => getGroups(dispatch).then(() => Actions.groups()));
+          postGroup(data)
```

Comment:
postGroup(data).then(data => {
               Actions.groups();
              getGroups(dispatch);
            });
postGroup(data).then(data => getGroups(dispatch).then(() => Actions.groups()))

the original has 2 thens in it. it waits till the getgroups finishes to assure all the groups are loaded before it redirects, the way you changed it, it redirects first, then gets the groups